# Generated by Django 2.2.6 on 2019-10-29 23:36

from django.conf import settings
from django.db import migrations, models
from django.core.exceptions import ObjectDoesNotExist, MultipleObjectsReturned


def relocate_owner_schema(apps, schema_editor):
    _flat = apps.get_model('property', 'flat')
    _owner = apps.get_model('property', 'owner')
    for flat in _flat.objects.all():
        _owner.objects.update_or_create(owner_name=flat.owner,
                                     owners_phonenumber=flat.owners_phonenumber,
                                     owner_phone_pure=flat.owner_phone_pure)

    for owner in _owner.objects.all():
        owners_all_flats = _flat.objects.filter(owner=owner.owner_name)
        for flat in owners_all_flats:
            owner.flats.add(flat)
            owner.save()
            print(f'{owner} владеет {owners_all_flats}')


# # Для отката - предыдущая миграция:
# python manage.py migrate property 0018_auto_20191117_1314
def move_backward(apps, schema_editor):
    _flat = apps.get_model('property', 'flat')
    _owner = apps.get_model('property', 'owner')
    for flat in _flat.objects.all():
        _owner.objects.filter(owner_name=flat.owner).delete()

    for owner in _owner.objects.all():
        owner.flats.set('')
        owner.save()
        print(f'move_backward {owner}')


class Migration(migrations.Migration):

    dependencies = [
        ('property', '0018_auto_20191117_1314'),
    ]

    operations = [
        migrations.RunPython(relocate_owner_schema, move_backward),
    ]
